### PR TITLE
Updated to include message body(media type) which is in the token.

### PR DESCRIPTION
**Pull request by KyleKayfish**
Updated to gracefully fail on exception when processing autoclose list

# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/AutoCloseSearchRequestJob.cs [hunk 1]
```diff
@@ -57,7 +57,15 @@ public async Task Execute(IJobExecutionContext context)
...
                     {
                         _logger.LogInformation("Autoclose: Calling ssg_SearchRequestCreateCouldNotAutoCloseNote SearchRequestId=" + ssgSearchRequest.SearchRequestId + " AutoCloseStatus "+ssgSearchRequest.AutoCloseStatus+"");
-                        await _searchRequestService.SearchRequestCreateCouldNotAutoCloseNote(ssgSearchRequest.SearchRequestId);
+                        try
+                        {
+                            await _searchRequestService.SearchRequestCreateCouldNotAutoCloseNote(ssgSearchRequest.SearchRequestId, cts.Token);
+                        }
+                        catch (Exception e)
+                        {
+                            _logger.LogInformation("Autoclose: Encountered Unexpected Exception");
```
Comment (pbolduc) [inline app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/AutoCloseSearchRequestJob.cs:66]: Why 2 log messages?  Isn't a single log call enough?

```cs
_logger.LogError(e, "Autoclose: Encountered Unexpected Exception");
```

I also noticed a lot of the other log messages you added in the past few weeks are not properly using semantic logging. For example,

```cs
 _logger.LogInformation("Autoclose: Successfully Processed " + requestList.Count + " Records");
```

should record the count as a property on the log message.

```cs
 _logger.LogInformation("Autoclose: Successfully Processed {SearchRequestCount} Records", requestList.Count);
```

File: app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/AutoCloseSearchRequestJob.cs [hunk 2]
```diff
@@ -57,7 +57,15 @@ public async Task Execute(IJobExecutionContext context)
                         ssgSearchRequest.AutoCloseStatus == SearchRequestAutoCloseStatusCode.CPMissingData.Value)
                     {
                         _logger.LogInformation("Autoclose: Calling ssg_SearchRequestCreateCouldNotAutoCloseNote SearchRequestId=" + ssgSearchRequest.SearchRequestId + " AutoCloseStatus "+ssgSearchRequest.AutoCloseStatus+"");
-                        await _searchRequestService.SearchRequestCreateCouldNotAutoCloseNote(ssgSearchRequest.SearchRequestId);
+                        try
+                        {
+                            await _searchRequestService.SearchRequestCreateCouldNotAutoCloseNote(ssgSearchRequest.SearchRequestId, cts.Token);
+                        }
+                        catch (Exception e)
```
Comment (pbolduc) [inline app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/AutoCloseSearchRequestJob.cs:64]: Are there know reasons?  Having a comment indicating what causes have been done and what the side effect of just skipping this record.  If the problem is not transient, could the number of search requests needing to be auto closed continue to increase until there are too many?

File: app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/AutoCloseSearchRequestJob.cs [hunk 3]
```diff
@@ -69,7 +77,16 @@ public async Task Execute(IJobExecutionContext context)
                                 { "statuscode", SearchRequestStatusCode.SearchRequestAutoClosed.Value}
                             };
                             _logger.LogInformation("Autoclose: Updating Search Status SearchRequestId=" + ssgSearchRequest.SearchRequestId + "");
-                            await _searchRequestService.UpdateSearchRequest(ssgSearchRequest.SearchRequestId, updatedFields, cts.Token);
+                            try
+                            {
+                                await _searchRequestService.UpdateSearchRequest(ssgSearchRequest.SearchRequestId, updatedFields, cts.Token);
+
```
Comment (pbolduc) [inline app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/AutoCloseSearchRequestJob.cs:83]: avoid extra whitespace

**Comment by pbolduc** [comment]
I know you were not the original author, however, on line 46 of `app\DynamicsAdapter\DynamicsAdapter.Web\SearchRequest\AutoCloseSearchRequestJob.cs` a `CancellationTokenSource` is created so you can pass it to various async methods further down. This should CancellationTokenSource should be removed from source and the `CancellationToken` from the `IJobExecutionContext` should be used.

**Review by pbolduc — CHANGES_REQUESTED** [review]

**Review by pbolduc — APPROVED** [review]

File: app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/AutoCloseSearchRequestJob.cs [hunk 4]
```diff
@@ -50,14 +50,21 @@ public async Task Execute(IJobExecutionContext context)
             {
                 _logger.LogInformation("Autoclose: Checking for Search Requests");
                 List<SSG_SearchRequest> requestList = await GetAutoCloseSearchRequestAsync(cts.Token);
-                _logger.LogInformation("Autoclose: Successfully Retrieved "+ requestList.Count+" Records");
+                _logger.LogInformation("Autoclose: Successfully Retrieved {RecordsCount}",+ requestList.Count);
```
Comment (NovaVic) [inline app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/AutoCloseSearchRequestJob.cs:53]: I think there is a leftover + or concatenation symbol that should be removed

**Review by NovaVic — COMMENTED** [review]
Minor cleanup in line 53 in app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/AutoCloseSearchRequestJob.cs would be good (there is extraneous concatenation ). Thanks!

**Pull request merged**